src/data_types/cred_request.rs

```diff
@@ -34,8 +34,10 @@ impl Validatable for CredentialRequest {
             }
             None => {
                 if self.cred_def_id.is_legacy_cred_def_identifier() {
-                    if let Some(prover_did) = self.prover_did.clone() {
-                        if LEGACY_DID_IDENTIFIER.captures(&prover_did).is_some() {
+                    if let Some(prover_did) = self.prover_did.as_deref() {
```

Comment:
Before, it would only accept an indy short DID such as 'DXoTtQJNtXtiwWaZAK3rB1'. Now it accepts anything URI-like as well, as indy-credx does (although credx requires two prefixes such as `did:indy:..`). I think if the client wants to use any freeform input then they should just use the entropy parameter.

Reply:
Just to confirm — URI-like as in `<letters and numbers>:<letter, numbers, :, / and the like>`?

Reply:
The pattern is `^[a-zA-Z][a-zA-Z0-9\+\-\.]*:.+$` - so a URI scheme consisting of a letter followed by zero or more letters, numbers, `+`, `-`, or `.`, followed by a colon, followed by one or more characters.